agent/src/com/jonnyzzz/teamcity/virtual/run/VMRunnerContext.java

```diff
@@ -68,22 +69,23 @@ public File getAgentTempDirectory() {
 
   @NotNull
   public String getCheckoutMountPoint() {
+    final String defaultMountPoint = SystemInfo.isUnix ? "/checkout" : "C:\\checkout";
     String mountPoint = myContext.getRunnerParameters().get(VMConstants.PARAMETER_CHECKOUT_MOUNT_POINT);
-    return StringUtil.isEmptyOrSpaces(mountPoint) ? "/checkout" : mountPoint;
+    return StringUtil.isEmptyOrSpaces(mountPoint) ? defaultMountPoint : mountPoint;
   }
 
   @NotNull
   public String getShellLocation() {
-    String loc = myContext.getRunnerParameters().get(VMConstants.PARAMETER_SHELL);
-    if (loc == null) {
+    final String loc = myContext.getRunnerParameters().get(VMConstants.PARAMETER_SHELL);
+    if (loc.equals("default")) {
       return "/bin/bash";
```

Comment:
In the code we have `VMRunnerContext` and tho child classes `DockerContext` and `VagrantContext`. Occasionally, we have a change e0bcf7ff220d1bd61a74e1fb86a7224dd04f6cc2 that added `getDockerMountMode` into the base class, and not into the `DockerContext`. 

I guess the best way to solve it is to move older methods to the `VagrantContext` and add new methods to the `DockerContext`.

It looks nice if we have a clear separation between Host and Container OS preferences (e.g. newlines, shell, paths). Currently we are lacking that, as you see